File: 0055-jump-game/0055-jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        int len = nums.length;
        int lastGoodIndex = len - 1;
        for (int i = len - 1; i >= 0; i--) {
            if (i + nums[i] >= lastGoodIndex) {
                lastGoodIndex = i;
            }
        }
        return lastGoodIndex == 0;        
    }
}

File: 0055-jump-game/0055-jump-game.rs
impl Solution {
    pub fn can_jump(nums: Vec<i32>) -> bool {
        let mut last_good_index = nums.len() - 1;
        for i in (0..nums.len()).rev() {
            if i + nums[i] as usize >= last_good_index {
                last_good_index = i;
            }
        }
        last_good_index == 0
    }
}
